forest/daemon/src/daemon.rs

```diff
@@ -394,7 +394,14 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
 async fn maybe_fetch_snapshot(should_fetch_snapshot: bool, config: Config) -> Config {
     if should_fetch_snapshot {
         let snapshot_path = default_snapshot_dir(&config);
-        let path = match snapshot_fetch(&snapshot_path, &config, is_aria2_installed()).await {
+        let path = match snapshot_fetch(
+            &snapshot_path,
+            &config,
+            &SnapshotServer::Filecoin,
```

Comment:
I believe by default we should grab our own snapshots for calibnet and only mainnet ones from an "external" source.

Reply:
yes this case is handled, by default we fetch from forest and only mainnet snapshot from filecoin

Reply:
@sudo-shashank this case is **not** handled, at least not entirely.

```
❯ target/release/forest --chain calibnet --target-peer-count 50 --encrypt-keystore false
 2022-12-01T10:02:47.304Z INFO  forest > Using default calibnet config
 2022-12-01T10:02:47.306Z INFO  forest::daemon > Starting Forest daemon, version 0.4.1+git.9c24be55
 2022-12-01T10:02:47.307Z INFO  forest_libp2p::service > Recovered libp2p keypair from "/home/rumcajs-work/.local/share/forest/libp2p/keypair"
 2022-12-01T10:02:47.307Z INFO  forest::daemon         > PeerId: 12D3KooWN9wV8DVwHXAFocGFJTpf7BwpTGkEE4nrtdJkBsf1f34X
 2022-12-01T10:02:47.307Z WARN  forest::daemon         > Warning: Keystore encryption disabled!
 2022-12-01T10:02:47.307Z INFO  forest::daemon         > Admin token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXSwiZXhwIjoxNjc1MDcyOTY3fQ.eSgUDci8kkJfBbsPE5JrHg_z3YzYP2G2E6Zd1q0RH3Q
 2022-12-01T10:02:47.346Z INFO  forest::daemon         > Prometheus server started at 0.0.0.0:6116
 2022-12-01T10:02:47.346Z WARN  forest_chain::store::chain_store > No previous chain state found
 2022-12-01T10:02:47.350Z INFO  forest_genesis                   > Initialized genesis: BlockHeader: Cid(bafy2bzacecyaggy24wol5ruvs6qm73gjibs2l2iyhcqmvi7r7a4ph7zx3yqd4)
✔ Forest needs a snapshot to sync with the network. Would you like to download one now? · yes
 2022-12-01T10:02:48.353Z INFO  forest::daemon                   > Using network :: calibnet
 2022-12-01T10:02:48.356Z INFO  forest_chain_sync::chain_muxer   > Evaluating network head...
 2022-12-01T10:02:48.356Z INFO  forest::daemon                   > JSON-RPC endpoint started at 127.0.0.1:1234
 2022-12-01T10:02:48.356Z INFO  forest_rpc                       > Ready for RPC connections
 2022-12-01T10:02:50.624Z INFO  forest_cli_shared::cli::snapshot_fetch > Snapshot url: https://pub-f851d666b9a54f60bd93202926618941.r2.dev/minimal/85200_2022_12_01T08_13_00Z.car
 2022-12-01T10:02:50.625Z INFO  forest_cli_shared::cli::snapshot_fetch > Snapshot will be downloaded to /home/rumcajs-work/.local/share/forest/snapshots/calibnet/filecoin_snapshot_calibnet_2022-12-01_height_85200.car
 2022-12-01T10:02:51.418Z INFO  forest_cli_shared::cli::snapshot_fetch > Expected sha256 checksum: eb41c8d8f77097c26dad0278d4d548df63bb9338abea557201d977b6810a9845

```